### PR TITLE
[Xcode] Add Menu Bar Command

### PR DESCRIPTION
**Pull request by SvenTiigi**
## Description

- Added `Show Recent Projects in Menu Bar` command with preferences allowing to specify the number of visible project and project types.
- Added search bar accessory to `Search Recent Projects` command `List` allowing to filter the results based on the selected project type.

## Screencast

Xcode-Menu-Bar-Command

Xcode-Menu-Bar-Preferences

Search-Recent-Projects-Filter

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/xcode/src/components/xcode-projects-menu-bar/xcode-projects-menu-bar.component.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+/**
+ * Xcode Projects Menu Bar
+ */
+export function XcodeProjectsMenuBar(): JSX.Element {
+  const { isLoading, data } = usePromise(XcodeProjectMenuBarService.xcodeProjects);
+  return (
+    <MenuBarExtra isLoading={isLoading} icon="xcode-menu-bar-icon.png" tooltip="Show Recent Xcode Projects">
+      <MenuBarExtra.Item title="Recent Xcode Projects" />
+      {data?.map((xcodeProject) => {
```
Comment (sxn) [inline extensions/xcode/src/components/xcode-projects-menu-bar/xcode-projects-menu-bar.component.tsx:14]: what about showing a `No recent projects` item if `data` is falsy and `isLoading` is `false`? 😄

File: extensions/xcode/src/components/xcode-projects-menu-bar/xcode-projects-menu-bar.component.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { XcodeProjectMenuBarService } from "../../services/xcode-project-menu-bar.service";
+
+/**
+ * Xcode Projects Menu Bar
+ */
+export function XcodeProjectsMenuBar(): JSX.Element {
+  const { isLoading, data } = usePromise(XcodeProjectMenuBarService.xcodeProjects);
+  return (
+    <MenuBarExtra isLoading={isLoading} icon="xcode-menu-bar-icon.png" tooltip="Show Recent Xcode Projects">
+      <MenuBarExtra.Item title="Recent Xcode Projects" />
```
Comment (sxn) [inline extensions/xcode/src/components/xcode-projects-menu-bar/xcode-projects-menu-bar.component.tsx:13]: very small suggestion - feel free to ignore though :D 
```suggestion
      <MenuBarExtra.Item title="Recent Xcode Projects" />
      <MenuBarExtra.Separator />
```

**Comment by sxn** [comment]
nice! I added two very small suggestions. You don't _have_ to implement them, but I think they'd improve UX ever so slightly. Once you've looked over them and decided, we should be good to go. :shipit:

**Comment by SvenTiigi** [comment]
Hi @sxn,

Thanks for your feedback ✌️

I have just added your recommendation to update the title based on the `isLoading` state as well as the availability of recent projects (https://github.com/raycast/extensions/pull/2598/commits/3f365532bea2268f8516837b250371ee61aaf4c6).

Additionally, I have added a `MenuBarExtra.Separator` if recent projects are available (https://github.com/raycast/extensions/pull/2598/commits/e189abe6711b23aa2fbd7d4389bf3560ef3601b0).

**Pull request merged**